### PR TITLE
[CPU] support QWen RoPE

### PR DESCRIPTION
**Pull request by usstq**
### Details:
 - *LLM QWen implemented rotary positional embedding in it's own way (other than gptneox or gptj or chatglm style), a new pattern and executor is added to support it.*

### Tickets:
 - *126990*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@luo-cheng2021 May I ask you to review the PR as well?

**Review by luo-cheng2021 — COMMENTED** [review]
LGTM.

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp [hunk 1]
```diff
@@ -574,6 +574,156 @@ ov::intel_cpu::RoPEFusionChatGLM::RoPEFusionChatGLM(int split_output_id) {
...
     };
 
+    auto m = std::make_shared<ngraph::pattern::Matcher>(result, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+ov::intel_cpu::RoPEFusionQwen::RoPEFusionQwen(int split_output_id) {
+    MATCHER_SCOPE(RoPEFusionQwen);
+
+    //if (std::getenv("NOQWEN")) return;
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp]: removed

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp [hunk 2]
```diff
@@ -574,6 +574,156 @@ ov::intel_cpu::RoPEFusionChatGLM::RoPEFusionChatGLM(int split_output_id) {
...
+
+    // since sequence length dimension of rotary_pos_emb has been sliced to past-kv-length + cur_len
+    // here the slicing `-cur_len:` returns the positions for current kv tokens
+    //
+    //       rotary_pos_emb = [i[:, -cur_len:, :, :] for i in rotary_pos_emb]
+    //
+    // given that StridedSlice on rotary_emb_cos/rotary_emb_sin is ensured to be the tail part
+    // and the result has compatible shape with query_proj/key_proj, we can safely assert that
+    // the length is the same as query_proj/key_proj, so we don't need to match it.
+    auto hidden_states = makePattern("f32[?,?,?]");  //
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp:607]: removed

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp [hunk 3]
```diff
@@ -574,6 +574,156 @@ ov::intel_cpu::RoPEFusionChatGLM::RoPEFusionChatGLM(int split_output_id) {
...
+            // key : split_output_id == 1
+            config.slice_start = config.head_cnt * config.head_size;
+            config.slice_stop = config.slice_start + config.head_cnt * config.head_size;
+        }
+
+        new_args.push_back(pattern_map.at(qkv_proj));
+        new_args.push_back(pattern_map.at(rotary_emb_cos));
+        new_args.push_back(pattern_map.at(rotary_emb_sin));
+
+        //std::cout << " RoPEFusionQwen: " << split_output_id << " : " << root << std::endl;
```
[the remark this answers is not in the document]
Comment (usstq) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp]: removed

**Review by luo-cheng2021 — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**